pkg/registry/bundlegraphloader.go

```diff
@@ -150,6 +153,37 @@ func (g *BundleGraphLoader) AddBundleToGraph(bundle *Bundle, graph *Package, ann
 		graph.Channels[channel] = channelGraph
 	}
 
+	// Remove the nodes that where patch-skipped previously, also in channels unaffected by the new bundle
+	if skippatch {
+		for channel := range graph.Channels {
+			channelGraph := graph.Channels[channel]
+			allSkips := make([]BundleKey, len(channelGraph.Nodes))
+			for node, replacedNodes := range channelGraph.Nodes {
+				for replaced := range replacedNodes {
+					nodeVersion, err := semver.Make(node.Version)
+					if err != nil {
+						// not semver, skip this replacement
+						continue
+					}
+
+					replacedVersion, err := semver.Make(replaced.Version)
+					if err != nil {
+						// not semver, skip this replacement
+						continue
+					}
+
+					if isSkipPatchCandidate(nodeVersion, replacedVersion) && len(channelGraph.Nodes[replaced]) == 0 {
+						allSkips = append(allSkips, replaced)
+					}
+				}
+			}
+
+			for _, node := range allSkips {
+				delete(channelGraph.Nodes, node)
+			}
+		}
+	}
+
```

Comment:
Super minor nit, but I think we can just reverse the conditional check and exit early, which will allow us to deflate the block iterating over the channels in the graph:

```go
if !skippath {
    return graph, nil
}

for channel := range graph.Channels {
    ...
}

return graph, nil
```